docker/docker_image_src_test.go

```diff
@@ -22,3 +23,30 @@ func TestSimplifyContentType(t *testing.T) {
 		assert.Equal(t, c.expected, out, c.input)
 	}
 }
+
+func toDockerRef(t *testing.T, ref string) dockerReference {
+	parsedRef, err := reference.ParseNamed(ref)
+	assert.Equal(t, nil, err)
+	return dockerReference{parsedRef}
+}
+
+func TestRewriteReferenceSuccess(t *testing.T) {
+	for _, c := range []struct{ ref, from, to string }{
+		{"example.com/image:latest", "example.com", "example.com"},
+		{"example.com/foo/image:latest", "example.com/foo", "example.com"},
+		{"example.com/image:latest", "image", "anotherimage"},
```

Comment:
This list is pretty short and only tests positive cases, which increases the chance to let potential regressions slip through the CI. We have a quite long list for the [prefix matching tests](https://github.com/containers/image/blob/master/pkg/sysregistriesv2/system_registries_v2_test.go#L129) that may serve a source of inspiration.

Reply:
Alright, I added more test cases which also contain the `docker.io/library` paths. Testing with `docker.io` only should not be needed since will be already covered by `reference.ParseNamed`.